V1/Services/StaySafeService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using CoviIDApiCore.Exceptions;
+using CoviIDApiCore.V1.Constants;
+using CoviIDApiCore.V1.DTOs.SafePlaces;
+using CoviIDApiCore.V1.Interfaces.Brokers;
+using CoviIDApiCore.V1.Interfaces.Repositories;
+using CoviIDApiCore.V1.Interfaces.Services;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CoviIDApiCore.V1.Services
+{
+    public class StaySafeService : IStaySafeService
+    {
+        private readonly IOrganisationAccessLogRepository _organisationAccessLogRepository;
+        private readonly ISafePlacesBroker _safePlacesBroker;
+
+        public StaySafeService(IOrganisationAccessLogRepository organisationAccessLogRepository, ISafePlacesBroker safePlacesBroker)
+        {
+            _organisationAccessLogRepository = organisationAccessLogRepository;
+            _safePlacesBroker = safePlacesBroker;
+        }
+
+
+
+        public async Task CaptureData(Guid walletId, DateTime twoWeeksFromDate)
+        {
+            var trails = new List<Trail>();
+            var logs = await _organisationAccessLogRepository.GetLogsForLastTwoWeeks(walletId, twoWeeksFromDate);
+
+            if (logs == null)
+                throw new NotFoundException(Messages.Oal_NotFound);
+
+            if (logs.Count > 0)
+            {
+                foreach (var log in logs)
+                {
+                    if (!(log.Latitude == 0 && log.Longitude == 0))
+                    {
+                        trails.Add(new Trail
+                        {
+                            Time = log.CreatedAt.ToString(),
+                            Latitude = decimal.ToDouble(log.Latitude),
+                            Longitude = decimal.ToDouble(log.Longitude)
+                        });
+                    }
+                }
+                if (trails.Count > 0)
+                {
+                    var request = new RedactedRequest
+                    {
+                        Identifier = Guid.NewGuid().ToString(),
```

Comment:
Do we maybe not need to record this somehow?

Reply:
I don't know, I thought about adding the wallet Id but do we want to know who was were or do we just want to know were a sick person was?